lib/src/protos/user_config.proto

```diff
@@ -0,0 +1,43 @@
+// Copyright 2025 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package user_config;
+
+// SecureUserConfig adds cryptographic signatures to ensure that any user
+// configuration was generated by the current user.
+// See https://github.com/jj-vcs/jj/issues/1595 for more details.
+message SecureUserConfig {
+  // Serialized UserConfig. Don't just put it as a submessage here because
+  // protobuf serialization is not canonical.
+  bytes storage = 1;
+
+  // This *probably* isn't necessary, but it doesn't hurt.
+  bytes salt = 2;
+  
+  // If I zip up a repository and send it to you, a valid signature is then
+  // known to all attackers. They can then send me a zip file including the
+  // same signature, and I would then have user configs enabled, despite the
+  // repo coming from someone else.
+  // We mitigate this risk by putting the path in as part of the signature.
+  string path = 3;
```

Comment:
This is an idea for future extension, but it's nice if we can trust config for path prefix such as `/home/user`. For example, suppose the user doesn't unzip untrusted repositories under `~/work`, they can set a config knob to sign their repo config files for `~/work`. Then they can move their repositories without re-signing configs.

Reply:
Keeping in mind the fact that this only occurs precisely when you move a repository that contains a custom `.jj/repo/config.toml`, this should be an extremely rare occurence.

I'm not a fan of trying to make optimizations to minimize the impact of this.